lib/recommendation.js

```diff
@@ -0,0 +1,61 @@
+var XMLHttpRequest = Cc['@mozilla.org/xmlextras/xmlhttprequest;1'].createInstance();
+
+
```

Comment:
You need EXPORTED_SYMBOLS here, see this MDN page for details: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/Using